### PR TITLE
CSV output for int and string reporter values

### PR DESCRIPTION
**Pull request by zachmprince**
Refs #16166

### CODE REVIEW

File: framework/include/outputs/TableOutput.h [hunk 1]
```diff
@@ -96,3 +100,47 @@ class TableOutput : public AdvancedOutput
   /// Enable/disable output of time column for Postprocessors
   const bool _time_column;
 };
+
+template <typename T>
+void
+TableOutput::outputReporter(const ReporterName & name)
```
Comment (aeslaughter) [inline framework/include/outputs/TableOutput.h:106]: I am thinking that this should be moved to the source file and have explicit instantiation of the supported types, this way you will see compiler options if something is added without support by child objects.

File: framework/include/outputs/TableOutput.h [hunk 2]
```diff
@@ -96,3 +100,47 @@ class TableOutput : public AdvancedOutput
...
+
+    const T & value = _reporter_data.getReporterValue<T>(name);
+    _reporter_table.addData<T>(name.getCombinedName(), value);
+    _all_data_table.addData<T>(name.getCombinedName(), value);
+  }
+}
+
+template <typename T>
+void
+TableOutput::outputVectorReporter(const ReporterName & name)
```
Comment (aeslaughter) [inline framework/include/outputs/TableOutput.h:126]: same here

File: framework/src/utils/FormattedTable.C [hunk 3]
```diff
@@ -50,6 +51,68 @@ dataLoad(std::istream & stream, FormattedTable & table, void * context)
...
+  value_base->store(stream, context);
+}
+
+template <>
+void
+dataLoad(std::istream & stream, std::shared_ptr<TableValueBase> & value_base, void * context)
+{
+  std::string type;
+  dataLoad(stream, type, context);
+  if (type == typeid(bool).name())
```
Comment (aeslaughter) [inline framework/src/utils/FormattedTable.C:67]: Look into std::is_same for these ifs

File: framework/src/outputs/TableOutput.C [hunk 4]
```diff
@@ -113,24 +113,50 @@ TableOutput::outputPostprocessors()
...
+    outputReporter<long long int>(r_name);
+    outputReporter<float>(r_name);
+    outputReporter<long double>(r_name);
+    outputReporter<char>(r_name);
+    outputReporter<char *>(r_name);
+    outputReporter<std::string>(r_name);
+
+    // Reals need to be last because of pps
+    if (!hasPostprocessorByName(r_name.getObjectName()))
+      outputReporter<Real>(r_name);
```
Comment (aeslaughter) [inline framework/src/outputs/TableOutput.C:140]: Why last? Isn't the if statement enough regardless of when in this list it is called?

**Review by aeslaughter — CHANGES_REQUESTED** [review]
Can you add unit tests for the new features of FormattedTable

https://github.com/idaholab/moose/blob/next/unit/src/FormattedTableTest.C

File: framework/include/utils/FormattedTable.h [hunk 5]
```diff
@@ -28,6 +30,98 @@ template <>
...
+{
+public:
+  virtual ~TableValueBase() = default;
+
+  template <typename T>
+  bool isSupportedType() const
+  {
+    return std::find(_supported_types.begin(), _supported_types.end(), typeid(T).name()) !=
+           _supported_types.end();
+  }
```
[the remark this answers is not in the document]
Comment (zachmprince) [reply framework/include/utils/FormattedTable.h:48]: That's a good idea. I figured people might want to add other types to output. But maybe they could just add to the `is_same` instead.

File: framework/include/outputs/TableOutput.h (hunk 1 above)
Comment (zachmprince) [reply framework/include/outputs/TableOutput.h:106]: Good idea

File: framework/src/outputs/TableOutput.C (hunk 4 above)
Comment (zachmprince) [reply framework/src/outputs/TableOutput.C:140]: I think you're right. I honestly forgot why I put this last.

File: framework/src/utils/FormattedTable.C (hunk 3 above)
Comment (zachmprince) [reply framework/src/utils/FormattedTable.C:67]: We need to store the type somehow has restartable data. My way was to just store the typeid string. Is there another way we can store the type?

Comment (zachmprince) [reply framework/src/utils/FormattedTable.C:67]: I think if we were able to use `is_same`, we wouldn't need these if statements.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**